pygfx/utils/show.py

```diff
@@ -34,33 +44,47 @@ def show(object: WorldObject, up=None):
 
         background = Background(None, BackgroundMaterial((0, 1, 0, 1), (0, 1, 1, 1)))
         scene.add(background)
+        scene.add(AmbientLight(0.2))
```

Comment:
I prefer the old version of determining whether to add lights. Imagine:
```py
cube = gfx.Mesh(...)
gfx.show(cube)
```
Works fine, now I modify my example to show two cubes:
```py
scene  = gfx.Scene()
scene.add(gfx.mesh(...))
scene.add(gfx.mesh(...))
gfx.show(scene)
```
and now it's all dark :)

Reply:
You could use a Group, but I agree that it's maybe surprising behavior

Reply:
The opposite holds true as well though "The scene works in `gfx.show`. Why don't I see anything in my actual setup!?".

It's meaningful to add lights if we only get an object and construct a default scene (users will likely place that object into their own scene later). However, when we get a full scene I think we should be mindful before modifying it; especially when we think of `gfx.show` as a debugging tool.

Reply:
How about just adding a kwarg `add_lights=True`

Reply:
> The opposite holds true as well though "The scene works in gfx.show. Why don't I see anything in my actual setup!?".

Good point. There is a chance of confusion either way.

What bothers me about `if isinstance(object, Scene)` is that it means the behavior of this function depends on whether a single object is passed, compared to multiple objects grouped in a scene. E.g. adding lights and background.

After some thought, I like @Korijn's suggestion about using `Group`: we can keep the behavior, but add a note in the docstring about the behavior when a `Scene` is given, and that a `gfx.Group()` should probably be used for multiple objects. And also use that in examples.

To take this idea further, if we create the camera, it should only be added to the scene (with a light attached) when we also create the scene here.

Reply:
I added a suggestion for my last point. Could you update the docs to explain `Scene` vs `Group`?

Reply:
I really do think an `add_lights=True` kwarg would be better, honestly. It's explicit, and obvious.

Reply:
One reason I was leaning towards using the type of the given object to decide whether to add stuff, was that we also background and a camera to it. Would it make sense to generalize  `add_lights=True` to something like `modify_scene=True`?

Reply:
How about: `add=('lights', 'camera', 'background', 'controller')`

I wish I could put `action` in there :)

Reply:
mmm, what about `add_lights` and `add_background` args? (The camera being added is by virtue of adding the directional light.)

Reply:
They're all equally acceptable to me, I just think it's better to lean towards controlling the behavior of `gfx.show()` via explicit keyword arguments than letting it be derived from the scene graph in a rather opinionated manner.

Reply:
Hm. At first, I was going to agree with `add_lights`, but when thinking about it a bit more I am not sure anymore if it solves the problem. 

Assuming `add_lights` has a default value like `add_lights=True` then chances are most people will use `gfx.show(scene)` and not know/care about most additional kwargs unless explicitly prompted to check. Especially if there are many. (In my experience, many devs only check the docs if something is broken and then only if SO doesn't have good suggestions.) So even though we can now turn this behavior off, it is still somewhat unintuitive and we are not exactly advertising it.
As such, the switch itself doesn't really solve the "Why does it work in `gfx.show`, but not in my application" problem.

We could, ofc, choose `add_lights=False` as a default but that would defeat the point of the kwarg. It's equal work to do `scene.add(gfx.SomeLight())` vs explicitly passing `add_lights=True`, so there is nothing gained. 

Also, in the case of us wrapping the input in a scene, we will likely have to do the lights and ignore this kwarg. This would make our kwargs quite complicated, because we already have `canvas`, `before_render`, and `after_render` that may be ignored in certain circumstances. That's quite a complex API for something that is meant to be a "quick scaffolding tool to debug your application or develop pygfx".

Reply:
Thinking of `gfx.show` as a diagnostics/debugging tool, I like the following:

If we don't get a `Scene`, we create one and add a light since this scene typically contains the ambient light. If we do get a `Scene`, we check if there are light sources and if not, we show a warning along the lines of `"Your scene does not contain a light source and will appear black."`. 

This way, there are no surprises. We can visualize single objects as expected, we don't modify existing scenes, and we give useful feedback on "ad-hoc" scenes like the one from @almarklein 's description. Would this work for you @almarklein ?

Reply:
Yeah, I think this could work 👍 . I'd say something like "... and some objects may not be visible".

Reply:
This is now implemented :)